test/functional/core/exit_spec.lua

```diff
@@ -43,4 +43,17 @@ describe('v:exiting', function()
     run(on_request, nil, on_setup)
   end)
 
+  it('is specified exit code after :cquit', function()
+    local function on_setup()
+      command('autocmd VimLeavePre * call rpcrequest('..cid..', "")')
+      command('autocmd VimLeave    * call rpcrequest('..cid..', "")')
+      command('cquit 123')
+    end
+    local function on_request()
+      eq(123, eval('v:exiting'))
+      return ''
+    end
+    run(on_request, nil, on_setup)
+  end)
```

Comment:
Missing negative checks: what will happen if you supply it non-number argument? Number followed by non-number? Out-of-range for exit codes (e.g. 1024 while maximal exit code is 255) number? Out-of-range for regular integer number?

(BTW, `COUNT` takes care of that too: `:cnext 1x` and `:cnext x` are errors and you are absolutely missing checking for errors what you must do.)